e2e/Dockerfile

```diff
@@ -0,0 +1,5 @@
+FROM golang:1.19
+
+WORKDIR /go/src/github.com/Azure/aks-app-routing-operator
+ADD ../. .
```

Comment:
If this runs after Terraform (my understanding is that it does) then the image will contain the test cluster's kubeconfig.

Can we pass any configuration necessary to run the tests through the test job via environment variables or a configmap mount? And use the in cluster config to reach apiserver instead of the kubeconfig.

Reply:
Oh- it would also be great to build the test binary in a build container and push only it rather than also pushing the source code.

Reply:
for the first bit - we can just remove the kubeconfig entirely - the containerized tester already uses the in cluster config, see [here](https://github.com/Azure/aks-app-routing-operator/pull/33/files#diff-e91d29c6d4c58503d9c2d72e2b04f1b70901a613e23008d1d1cfe8beff7038d4R47).

As far as I can tell, the only config that the e2e config needs is the e2e.json file. How would you feel about a go-embed of that JSON as opposed to creating a configmap? That way we're not importing extra stuff into the binary that just gets built once, but saving the step of having to write that info into a configmap and then deploying that

Reply:
actually - looks like because of the file structure, embed wouldn't be the cleanest solution. working on using a configmap now.